### PR TITLE
update

### PR DESCRIPTION
**Pull request by VTui22**

### CODE REVIEW

**Comment by VTui22** [comment]
fix skin mesh position

File: private/Producers/FbxProducer/FbxProducerImpl.cpp [hunk 1]
```diff
@@ -59,9 +59,14 @@ cd::Vec3f ConvertFbxScale(const fbxsdk::FbxVector4& scale)
 	return cd::Vec3f(static_cast<float>(scale[0]), static_cast<float>(scale[1]), static_cast<float>(scale[2]));
 }
 
-cd::Transform ConvertFbxTransform(const fbxsdk::FbxVector4& position, const fbxsdk::FbxQuaternion& rotation, const fbxsdk::FbxVector4& scale)
+cd::Transform ConvertFbxTransform(fbxsdk::FbxAMatrix Matrix)
```
Comment (T-rvw) [inline private/Producers/FbxProducer/FbxProducerImpl.cpp]:
```suggestion
cd::Transform ConvertFbxTransform(const fbxsdk::FbxAMatrix& Matrix)
```
Avoid copy when passing matrix parameter.

File: private/Producers/FbxProducer/FbxProducerImpl.cpp [hunk 2]
```diff
@@ -59,9 +59,14 @@ cd::Vec3f ConvertFbxScale(const fbxsdk::FbxVector4& scale)
...
 {
-	return cd::Transform(ConvertFbxPosition(position), ConvertFbxRotation(rotation), ConvertFbxScale(scale));
+	cd::Matrix4x4 transformMatrix = ConvertFbxMatrixToCDMatrix(Matrix);
+
+	return cd::Transform(
+		cd::Vec3f(transformMatrix.GetTranslation()),
+		cd::Quaternion(cd::Quaternion::FromMatrix(transformMatrix.GetRotation())),
```
Comment (T-rvw) [inline private/Producers/FbxProducer/FbxProducerImpl.cpp]:
```suggestion
		cd::Quaternion::FromMatrix(transformMatrix.GetRotation()),
```
Already `cd::Quaternion `type. No need to cast.

File: private/Producers/FbxProducer/FbxProducerImpl.cpp [hunk 3]
```diff
@@ -1392,26 +1397,20 @@ void FbxProducerImpl::ImportSkeletonBones(fbxsdk::FbxScene* pScene, const std::v
...
 		}
 		else
 		{
 			// RootBone is already in global coordinate system.
-			const auto& rootTransform = globalTransformPerLinkBone[cdBoneID];
-			localLinkT = rootTransform.GetT();
-			localLinkQ = rootTransform.GetQ();
-			localLinkS = rootTransform.GetS();
+			const auto& transform = globalTransformPerLinkBone[cdBoneID];
+
```
Comment (T-rvw) [inline private/Producers/FbxProducer/FbxProducerImpl.cpp]: Defined variable `transform` twice.
Also, you should consider `boneIndex != rootBoneIndex`. Please have a check if `transform` is an identity matrix.
```suggestion
			transform = globalTransformPerLinkBone[cdBoneID];
		
```

File: private/Producers/FbxProducer/FbxProducerImpl.cpp [hunk 4]
```diff
@@ -1392,26 +1397,20 @@ void FbxProducerImpl::ImportSkeletonBones(fbxsdk::FbxScene* pScene, const std::v
...
-			const auto& rootTransform = globalTransformPerLinkBone[cdBoneID];
-			localLinkT = rootTransform.GetT();
-			localLinkQ = rootTransform.GetQ();
-			localLinkS = rootTransform.GetS();
+			const auto& transform = globalTransformPerLinkBone[cdBoneID];
+		
 		}
 
-		cdBone.SetTransform(details::ConvertFbxTransform(localLinkT, localLinkQ, localLinkS));
+		cdBone.SetTransform(details::ConvertFbxTransform(transform));
```
Comment (T-rvw) [inline private/Producers/FbxProducer/FbxProducerImpl.cpp]: `transform` is not a good name.`boneTransform` is better.

File: private/Producers/FbxProducer/FbxProducerImpl.cpp [hunk 5]
```diff
@@ -1392,26 +1397,20 @@ void FbxProducerImpl::ImportSkeletonBones(fbxsdk::FbxScene* pScene, const std::v
...
 		}
 		else
 		{
 			// RootBone is already in global coordinate system.
-			const auto& rootTransform = globalTransformPerLinkBone[cdBoneID];
-			localLinkT = rootTransform.GetT();
-			localLinkQ = rootTransform.GetQ();
-			localLinkS = rootTransform.GetS();
+			boneTransform = globalTransformPerLinkBone[cdBoneID];
+
```
Comment (T-rvw) [inline private/Producers/FbxProducer/FbxProducerImpl.cpp]:
```suggestion
```

File: private/Producers/FbxProducer/FbxProducerImpl.cpp [hunk 6]
```diff
@@ -1392,26 +1397,20 @@ void FbxProducerImpl::ImportSkeletonBones(fbxsdk::FbxScene* pScene, const std::v
...
-		fbxsdk::FbxVector4 localLinkS;
+		fbxsdk::FbxAMatrix boneTransform;
 		if (boneIndex != rootBoneIndex)
 		{
-			fbxsdk::FbxAMatrix transform = globalTransformPerLinkBone[cdBoneParentID].Inverse() * globalTransformPerLinkBone[cdBoneID];
-			localLinkT = transform.GetT();
-			localLinkQ = transform.GetQ();
-			localLinkS = transform.GetS();
+			boneTransform = globalTransformPerLinkBone[cdBoneParentID].Inverse() * globalTransformPerLinkBone[cdBoneID];
+
```
Comment (T-rvw) [inline private/Producers/FbxProducer/FbxProducerImpl.cpp]:
```suggestion
```

**Review by T-rvw — APPROVED** [review]

**Comment by T-rvw** [comment]
Resolve conversations if already fixed.
Note to use `Squash and Merge` if the fix is good to merge after testing.

**Comment by T-rvw** [comment]
Merge commit should have a more meaningful comment rather than "update" and delete remote branch if you don't need it again.

**Pull request merged**